server/src/main/java/org/jboss/as/server/ServerEnvironmentService.java

```diff
@@ -54,9 +55,14 @@
  */
 public class ServerEnvironmentService implements Service<ServerEnvironment> {
 
+
     /** Standard ServiceName under which a ServerEnvironmentService would be registered */
+    /** @deprecated Use {@link #SERVER_ENVIRONMENT_SERVICE_NAME instead} */
+    @Deprecated
     public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append("server", "environment");
 
+    public static final ServiceName SERVER_ENVIRONMENT_SERVICE_NAME = SERVER_ENVIRONMENT_CAPABILITY.getCapabilityServiceName();
```

Comment:
We don't want to create yet another public ServiceName.  Consumers are meant to acquire the ServerEnvironment via a capability requirement (using its name).

Reply:
In that case, capability should be public. It is the default right now. 
For example here https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java#L148

Reply:
@pferraro I guess the issue without using a service name created from the capability is that it is required a capability service target to use the capability name for the server environment service. 

That's the reason I gave an LGTM as it is now, we have used the same approach for several services created from the capabilities, see for example the [MANAGEMENT_EXECUTOR](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/ServerService.java#L160) at ServerService.

Maybe there is another way to do it, and if there is one, is unknown to me.

Reply:
@pferraro , right!, I missed that link, thanks.

Reply:
@pferraro I misunderstood the approach and made a mistake with my previous comment (thanks for ping me @parsharma). We cannot access the RuntimeCapability from [HttpManagementAddHandler.java#L148](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java#L148) because the capability was requested to be declared with package visibility. 

If we do not expose the service name publicly in [ServerEnvironmentService.java](https://github.com/wildfly/wildfly-core/pull/5480/files/70d5b038e370e5029b8f07b683b423d2fed02bde#diff-1e87d3842cfaee888f1f6c3a1aa1d55132dd410d7577cfeb60179637f9e39dba), I understand we have to declare the capability with public visibility at ServerService so we can get the service name in HttpManagementAddHandler.java#L148 and for any service target that cannot manage capability names. Otherwise, if there is any other approach, please, describe it here since we do not see it yet.

Reply:
The logic in [HttpManagementAddHandler](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java#L148) is incorrect.  See: https://github.com/wildfly/wildfly-core/pull/5516/commits/c9580133c7984edf3d1edaeeebd51d57ea4f8d8f
... taken from https://github.com/wildfly/wildfly-core/pull/5516 which incorporates this PR.

Perhaps it would best to pull that change into a separate JIRA.  I'll create a pull request for that momentarily, then @parsharma can base her work on that branch.

Reply:
@pferraro I have one more doubt here https://github.com/wildfly/wildfly-core/pull/5480/files#diff-700a2581dad9cd3005f25b8c3a70acc323cd43d385ed8f8014a52b01a557605dR68 we are using ServiceBuilder so we need ServiceName. Do you think we need to change it to capabiltyBuilder?

Reply:
> @pferraro I have one more doubt here https://github.com/wildfly/wildfly-core/pull/5480/files#diff-700a2581dad9cd3005f25b8c3a70acc323cd43d385ed8f8014a52b01a557605dR68 we are using ServiceBuilder so we need ServiceName. Do you think we need to change it to capabiltyBuilder?

Normally, ManagedExplodedContentServitor would indeed use the service target returned via OperationContext.getCapabilityServiceTarget(), however the ServiceBuilder currently returned via CapabilityServiceTarget.addService(ServiceName) does not support capability resolution.  This is being addressed by https://github.com/wildfly/wildfly-core/pull/5588
Until then, you can generate the requisite ServiceName via OperationContext.getCapabilityServiceName("org.wildfly.server.environment")